charms/worker/k8s/src/charm.py

```diff
@@ -618,6 +670,39 @@ def _on_update_status(self, _event: ops.UpdateStatusEvent):
         except status.ReconcilerError:
             log.exception("Can't update_status")
 
+    def kubectl(self, *args) -> str:
```

Comment:
The charm file starts to get quite large. Maybe it makes sense to split it up in the future, e.g. by moving such utils into their own classes.

Reply:
it is -- and i got REALLY close to pulling the trigger on that here.